packages/dds/tree/src/simple-tree/jsonSchema.ts

```diff
@@ -0,0 +1,208 @@
+	 * This is used to disambiguate the type of {@link TreeNodeSchema} this JSON Schema maps to.
+	 *
+	 * Note: This property name is not a part of the JSON Schema spec. This is a Fluid-specific extension.
+	 */
+	readonly _kind: TNodeKind;
+
+	/**
+	 * {@inheritDoc JsonSchemaType}
+	 */
+	readonly type: TJsonSchemaType;
+}
+
+/**
+ * JSON Schema for an object node.
+ * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-instance-data-model}.
+ * @alpha
+ */
+export interface ObjectNodeJsonSchema extends NodeJsonSchemaBase<"object", "object"> {
+	/**
+	 * Object fields.
+	 * @remarks Required fields should have a corresponding entry in {@link ObjectNodeJsonSchema.required}.
+	 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-properties}.
+	 */
+	readonly properties: Record<string, FieldJsonSchema>;
+
+	/**
+	 * List of keys for required fields.
+	 *
+	 * @remarks
+	 * Optional fields should not be included in this list.
+	 * Each key specified must have an entry in {@link ObjectNodeJsonSchema.properties}.
+	 */
+	readonly required?: string[];
+
+	/**
+	 * Whether or not additional properties (properties not specified by the schema) are allowed in objects of this type.
+	 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-additionalproperties}.
+	 */
+	readonly additionalProperties?: boolean;
+}
+
+/**
+ * JSON Schema for an array node.
+ * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-instance-data-model}.
+ * @alpha
+ */
+export interface ArrayNodeJsonSchema extends NodeJsonSchemaBase<"array", "array"> {
+	/**
+	 * The kinds of items allowed under the array.
+	 * @remarks Always represented via references to {@link TreeJsonSchema.$defs}.
+	 *
+	 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-items}.
+	 */
+	readonly items: {
+		/**
+		 * The kinds of items allowed under the array.
+		 * @remarks Always represented via references to {@link TreeJsonSchema.$defs}.
+		 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-anyof}.
+		 */
+		anyOf: JsonSchemaRef[];
+	};
+}
+
+/**
+ * JSON Schema for a map node.
+ * @remarks Special case for map nodes, which do not have a native JSON schema corollary.
+ * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-instance-data-model}.
+ * @alpha
+ */
+export interface MapNodeJsonSchema extends NodeJsonSchemaBase<"map", "object"> {
+	/**
+	 * Used to control the types of properties that can appear in the "object" representation of the map.
+	 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-patternproperties}.
+	 */
+	readonly patternProperties: {
+		/**
+		 * Types allowed in the map.
+		 * @remarks This format allows for any (JSON-compliant) key, but restricts the allowed types to only those specified.
+		 */
+		"^.*$": FieldJsonSchema;
```

Comment:
Is it possible to construct a regex that includes all strings except for `_kind`? Would that be helpful?

Reply:
`_kind` isn't something I would expect us to special case on the _data_. It's something we annotate the _schema_ with. Once we develop a solution to the general problem of disambiguating ambiguous input data, we may need to account for that here. But for now, `_kind` is only expected (specially) on the _schema_.